python/cudaq/kernel/ast_bridge.py

```diff
@@ -629,62 +777,65 @@ def getStructMemberIdx(self, memberName, structTy):
             )
         return structIdx, mlirTypeFromPyType(userType[memberName], self.ctx)
 
+    def __copyStructAndConvertElements(self, struct, expectedTy = None, conversion=None):
```

Comment:
nit: a code comment here, especially for the `conversion` callback argument, might be helpful.